src/ascii/response/check.rs

```diff
@@ -62,11 +63,58 @@ impl<R: Response, F: Fn(R) -> Result<R, AsciiCheckError<R>>> Check<R> for F {
     }
 }
 
+/// An implementation detail used by [`Port::set_default_response_check`](crate::ascii::Port::set_default_response_check).
+///
+/// This is a helper type that users should never need to use directly. It's
+/// primary purpose is to facilitate the conversion of any type that implements
+/// `Check<R: Response>` to a new type that implements `Check<AnyResponse>`.
+#[derive(Debug)]
+pub struct DefaultResponseCheck<K, R>(K, std::marker::PhantomData<R>);
```

Comment:
To avoid making the API documentation, consider moving this into a `details` mod.

Reply:
Consider renaming this to something representing its function rather than where it is used. Something like `AnyResponseCheck`.

Reply:
Maybe include trait bounds on the type - it may simplify the bounds needed on the port method.

Reply:
Did not create `details` mod - it seemed excessive. But did rename it to `AnyResponseCheck`. Adding trait bounds to the type didn't help, though the trait bounds could already be simplified on the port method.